lib/matplotlib/colors.py

```diff
@@ -1947,3 +1950,38 @@ def from_levels_and_colors(levels, colors, extend='neither'):
 
     norm = BoundaryNorm(levels, ncolors=n_data_colors)
     return cmap, norm
+
+
+def shade_color(color, percent):
```

Comment:
In almost all of the other color code we handle fractions as floats in range `[0, 1]`, should do the same here.